pkg/daemon/daemon.go

```diff
@@ -572,14 +572,23 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 	// or there is a new config we need to apply
 	// When using systemd configuration we write the file
 	if dn.useSystemdService {
-		r, err := systemd.WriteConfFile(latestState, dn.devMode, dn.platform)
+		systemdConfModified, err := systemd.WriteConfFile(latestState, dn.devMode, dn.platform)
```

Comment:
This is a great opportunity to clean this up using https://github.com/coreos/go-systemd

Can you refactor and get rid of all the manual systemd calls? The above library is the official one. That would mean we can clean up pkg/systemd/systemd.go by using that lib.

Reply:
Hi @bn222 that package is great does it only work for RHEL/COREOS or also other operating systems?

Reply:
@bn222  can you open an issue for this ? i do not want to turn this bug fix into a refactor.

Reply:
I agree that this is a good idea to switch to the library, but let's handle this as a separate PR with refactoring.

Reply:
@SchSeba I believe it works in general. 
@adrianchiris The downside of having a separate PR is that it tends to get forgotten. I've opened https://github.com/k8snetworkplumbingwg/sriov-network-operator/issues/533, so let's not let it linger for too long.